### PR TITLE
Devtooling-182: Refactoring Responsemanagement ResponseAsset

### PR DESCRIPTION
**Pull request by BrianMoyles**

### CODE REVIEW

File: genesyscloud/responsemanagement_responseasset/genesyscloud_responsemanagement_responseasset_proxy.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+// deleteRespManagementRespAsset deletes a Genesys Cloud responsemanagement responseasset by Id
+func (p *responsemanagementResponseassetProxy) deleteRespManagementRespAsset(ctx context.Context, id string) (response *platformclientv2.APIResponse, err error) {
+	return p.deleteRespManagementRespAssetAttr(ctx, p, id)
+}
+
+// createRespManagementRespAssetFn is an implementation of the function to create a Genesys Cloud responsemanagement responseasset
+func createRespManagementRespAssetFn(ctx context.Context, p *responsemanagementResponseassetProxy, respAsset *platformclientv2.Createresponseassetrequest) (*platformclientv2.Createresponseassetresponse, int, error) {
+	postResponseData, resp, err := p.responseManagementApi.PostResponsemanagementResponseassetsUploads(*respAsset)
+	if err != nil {
+		return nil, 0, fmt.Errorf("Failed to upload response asset: %v", err)
```
Comment (dginty4) [inline genesyscloud/responsemanagement_responseasset/genesyscloud_responsemanagement_responseasset_proxy.go:80]: You need to return `resp.StatusCode` here. Retry logic is based on the status code

File: genesyscloud/responsemanagement_responseasset/genesyscloud_responsemanagement_responseasset_proxy.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+		return nil, 0, fmt.Errorf("Failed to update Responsemanagement response asset %s: %v", id, err)
+	}
+	return putResponseData, resp.StatusCode, nil
+}
+
+// getRespManagementRespAssetByIdFn is an implementation of the function to get a Genesys Cloud responsemanagement responseasset by Id
+func getRespManagementRespAssetByIdFn(ctx context.Context, p *responsemanagementResponseassetProxy, id string) (*platformclientv2.Responseasset, *platformclientv2.APIResponse, error) {
+	sdkAsset, resp, getErr := p.responseManagementApi.GetResponsemanagementResponseasset(id)
+	if getErr != nil {
+		return nil, nil, fmt.Errorf("failed to retrieve response asset: %s", getErr)
```
Comment (dginty4) [inline genesyscloud/responsemanagement_responseasset/genesyscloud_responsemanagement_responseasset_proxy.go:98]: Same here, return the resp when there's an error. Same for the update and delete

**Review by charliecon — APPROVED** [review]

File: genesyscloud/responsemanagement_responseasset/resource_genesyscloud_responsemanagement_responseasset_schema.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+			"name": {
+				Description: "Response asset name.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+		},
+	}
+}
+
+func GenerateResponseManagementResponseAssetResource(resourceId string, fileName string, divisionId string) string {
```
Comment (dginty4) [inline genesyscloud/responsemanagement_responseasset/resource_genesyscloud_responsemanagement_responseasset_schema.go:72]: Put this in the resource file instead

**Comment by dginty4** [comment]
I think you forgot to update the reference in tf_export. Also Johns PR has caused a lot of merge conflicts. Once you resolve them it should be good to merge

**Review by dginty4 — APPROVED** [review]

**Pull request merged**